python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -605,3 +624,280 @@ def makeDataRefList(self, namespace):
             )
             self.refList.append(dataRef)
 
+def countMaskFromFootprint(mask,footprint, bitmask, ignoreMask):
+    """Function to count the number of pixels with a specific mask in a footprint
+    """
+    bbox = footprint.getBBox()
+    bbox.clip(mask.getBBox(afwImage.PARENT))
+    fp = afwImage.MaskU(bbox)
+    subMask = mask.Factory(mask, bbox, afwImage.PARENT)
+    afwDet.setMaskFromFootprint(fp, footprint, bitmask)
+    return numpy.logical_and((subMask.getArray() & fp.getArray()) > 0,
+                             (subMask.getArray() & ignoreMask) == 0).sum()
+
+
```

Comment:
I agree with this in principle. In practice, I'm not sure it's any less documented than the old `AssembleCoaddTask` (neither of them appears in [the Doxygen task list](https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/group___l_s_s_t__task__documentation.html) for example. I don't properly documenting everything is a prerequisite for merging given that this is an HSC port, but we do need a "technical debt" issue to record that it needs fixing in future.